### ISSUE TITLE
hidden-states from final (or middle layers)

### ISSUE THREAD

**Issue by janphilippfranken** [vllm-project/vllm#5406]
### 🚀 The feature, motivation and pitch

I am trying to extract hidden states from the final layer of llama3-8b (i.e., the final batch_size, seq_length, n_emb vector _before_ computing the logits). Would it be possible to add this functionality (i.e., access to hidden states similar to transformers ouput_hidden_states)? Thank you!

### Alternatives

HuggingFace Transformers, but this is too slow.

### Additional context

I am trying to train a SAE/linear probe on hidden states from llama3.

**Comment by jeejeelee** [comment]
Maybe you could consider utilizing nn.Module's hook.

**Comment by invisifire** [comment]
sample issue

**Comment by hmellor** [comment]
Closing as stale. The comment that triggered unstale doesn't appear to be constructive.
